torch/distributed/distributed_c10d.py

```diff
@@ -1320,6 +1320,68 @@ def gather_object(obj, object_gather_list=None, dst=0, group=group.WORLD):
         object_gather_list[i] = _tensor_to_object(tensor, tensor_size)
 
 
+def broadcast_object(obj, src, group=group.WORLD):
+    """
+    Broadcasts picklable object ``obj`` to the whole group. Similar to
+    :func:`broadcast`, but Python objects can be passed in. Note that the object
+    must be picklable in order to be broadcasted.
+
+    Arguments:
+        obj (Any): Input object to broadcast. Must be picklable. This argument
+            is ignored on ranks which are not the ``src`` rank.
+        src (int): Source rank from which to broadcast ``obj``.
+        group: (ProcessGroup, optional): The process group to work on.
+
+    Returns:
+        ``None``, if rank is not part of ``group``. Otherwise, will return the
+        broadcasted object.
+
+    .. note:: Note that this API differs slightly from the broadcast collective
+        since it does not provide an ``async_op`` handle and thus will be a
+        blocking call, and returns the object instead of setting the value in
```

Comment:
Question, will it be reasonable to take an `object_list` instead of `object`, so that we can populate the result there? I am little concern about returning the object here, because we might add async_op support later which will make this API return a Future. So, if we return the result object, future changes might bring BC issues. 

If we decided to use `object_list` as the argument, let's also rename the API to `broadcast_object_list`.

Reply:
Sure, so if I understand the semantics will be:
Src rank calls broadcast_object_list([obj], ...)
Other ranks call broadcast_object_list([None], ...)

And after the call the passed in list would have the broadcasted object at index 0? And in general, we wouldn't have to limit this to size 1 lists, it can be any size.

Reply:
Yep. For the non-source ranks, we can ignore all existing contents in the list. So it does not matter what are in the list when the API is called, we can always clear the list and then populate with results.